docs/zkapps/o1js/gadgets.mdx

```diff
@@ -0,0 +1,100 @@
+---
+title: Gadgets
+hide_title: true
+description: How gadgets work in o1js.
+keywords:
+  - smart contracts
+  - zkapp
+  - gadgets
+  - bitwise
+  - o1js
+  - blockchain
+---
+
+:::info
+
+Please note that zkApp programmability is not yet available on Mina Mainnet, but zkApps can now be deployed to Berkeley Testnet.
+
+:::
+
+# Gadgets
+
+Gadgets are amalgamations of the lowest level gates in the proof system (addition, multiplication) into more sophisticated boolean operations (XOR, NOT). Gadgets in o1js simplify the process of creating new cryptographic primitives and make it possible to implement other crypto algorithms, such as SHA. 
```

Comment:
I know I wrote this but I think it needs to be a bit more general.

Gadgets are amalgamations of the lowest level gates in the proof system (addition, multiplication) into more sophisticated operations. Gadgets in o1js simplify the process of creating new cryptographic primitives and make it possible to implement new crypto algorithms, such as SHA.

Reply:
@garwalsh  can you help me understand this feedback? 
I am not clear what to do with your comment on line 22... 
More general intro about gadgets, you mean? Let's talk so I understand

When we write our code, it turns into gates. Gates are like transistors. Gadgets are built out of gates, but not a program that a user would write.
Gates are configured to perform efficiently.
Gadgets are ... what in o1js?

Reply:
@barriebyron 
Will use suggestions in future. For this particular piece of feedback I have provided a slightly edited paragraph to replace the one starting at line 22